From 13691614d266257f29dcf0f6b0514357189cd9cb Mon Sep 17 00:00:00 2001
From: Brad Ebinger <breadley@google.com>
Date: Wed, 13 Jun 2018 17:18:20 -0700
Subject: [PATCH 02/39] Revert isResolvingImsBinding code in ImsResolver

When querying a carrier ImsService for IMS features
after a new SIM is inserted, there is an indeterminate period
of time where we do not know what features are supported.
During this time telephony gets an independent
CARRIER_CONFIG_CHANGED signal to update the IMS config to the
modem. We were attempting to supress this signal while
determining in the ImsResolver which ImsService MMTEL will
be created on. This resulted in a race condition when the
carrier ImsService only supported the RCS feature because
the carrier config update would be suppressed and the
ImsPhoneCallTracker would never send a config update (because
the device ImsService is still the only MMTEL feature).

Therefore this change reverts the code that detected this
since it was likely that the CARRIER_CONFIG_CHANGED signal
would be processed in other parts of telephony before it
was even processed here.

Bug: 109762920
Test: Telephony GTS/Unit Tests
Change-Id: I20a713b8858931dce2a304fc306bfe8553546f51
Merged-In: I20a713b8858931dce2a304fc306bfe8553546f51
---
 .../android/internal/telephony/ims/ImsResolver.java   | 11 -----------
 1 file changed, 11 deletions(-)

diff --git a/src/java/com/android/internal/telephony/ims/ImsResolver.java b/src/java/com/android/internal/telephony/ims/ImsResolver.java
index f0aee0141..4a030b179 100644
--- a/src/java/com/android/internal/telephony/ims/ImsResolver.java
+++ b/src/java/com/android/internal/telephony/ims/ImsResolver.java
@@ -1131,17 +1131,6 @@ public class ImsResolver implements ImsServiceController.ImsServiceControllerCal
         }
     }
 
-    /**
-     * @return true if the ImsResolver is in the process of resolving a dynamic query and should not
-     * be considered available, false if the ImsResolver is idle.
-     */
-    public boolean isResolvingBinding() {
-        return mHandler.hasMessages(HANDLER_START_DYNAMIC_FEATURE_QUERY)
-                // We haven't processed this message yet, so it is still resolving.
-                || mHandler.hasMessages(HANDLER_DYNAMIC_FEATURE_CHANGE)
-                || mFeatureQueryManager.isQueryInProgress();
-    }
-
     private String printFeatures(Set<ImsFeatureConfiguration.FeatureSlotPair> features) {
         StringBuilder featureString = new StringBuilder();
         featureString.append("features: [");
-- 
2.17.1

